cron_ext/extension.py

```diff
@@ -0,0 +1,232 @@
+                yield entry  # Preserve full-line comments
+            else:
+                match = cls.entry_pattern.fullmatch(entry)
+                if not match:
+                    continue
+                elif match.group("name") not in seen:
+                    seen.add(match.group("name"))
+                    yield entry
+
+    @property
+    def entries(self) -> tuple[str]:
+        lines = self.crontab.splitlines()
+        try:
+            a, b = self._find_meltano_section_indices(lines)
+        except ValueError:
+            return ()
+        return tuple(
+            line
+            for line in lines[a:b]
+            if not self.full_line_comment_pattern.fullmatch(line)
+        )
+
+    @entries.setter
+    def entries(self, new_entries: Iterable[str]) -> None:
+        lines = self.crontab.splitlines()
+        try:
+            a, b = self._find_meltano_section_indices(lines)
+        except ValueError:
+            # create new Meltano section at the bottom of the crontab
+            new_lines = (
+                *lines,
+                "",
+                "# ----- BEGIN MELTANO CRONTAB SECTION -----",
+                *new_entries,
+                "# ------ END MELTANO CRONTAB SECTION ------",
+                "",
+            )
+        else:
+            # update the existing Meltano section
+            new_lines = (
+                *lines[:a],
+                *new_entries,
+                *lines[b:],
+            )
+        self.crontab = "\n".join(new_lines)
+
+    def _new_entries(self, predicate: Callable[[str], bool]) -> Iterable[str]:
+        base_env = {"PATH": f'{os.environ["PATH"]}:$PATH'}
+        if "VIRTUAL_ENV" in os.environ:
+            base_env["VIRTUAL_ENV"] = os.environ["VIRTUAL_ENV"]
+        for schedule in self.meltano_schedule["elt"] + self.meltano_schedule["job"]:
+            interval = (
+                schedule["cron_interval"] if "job" in schedule else schedule["interval"]
+            )
+            args = " ".join(
+                schedule["job"]["tasks"] if "job" in schedule else schedule["elt_args"]
+            )
+            cmd = "run" if "job" in schedule else "elt"
+            env = {**base_env, **schedule["env"]}
+            env_str = " ".join(f'{key}="{value}"' for key, value in env.items())
+            if not predicate(schedule["name"]):
+                continue
+            command_script = self.cron_ext_dir / f"{schedule['name']}.sh"
+            command_script.write_text(
+                f"(cd '{self.meltano_project_dir}' && {env_str} "
+                f"meltano {cmd} {args}) 2>&1 | /usr/bin/logger -t meltano\n"
+            )
+            # Make the command script executable by the user who owns it
+            command_script.chmod(command_script.stat().st_mode | stat.S_IXUSR)
+            yield f"{interval} '{command_script.as_posix()}'"
+
+    def install(self, schedule_ids: list[str]) -> None:
```

Comment:
fyi - on macos this fails when the user has no crontab:

```
(ext-dev) ➜  extdev meltano invoke cron install
Unable to list crontab entries.
(ext-dev) ➜  extdev crontab -l
crontab: no crontab for syn
```

Reply:
Nice catch. I wasn't able to reproduce that, so it might be macOS specific. Should be fixed now.